### PR TITLE
inductor:enable conv+unary fusion for torch unary function

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #91677
* __->__ #91609

This PR is about to enable unary fusion which the unary is the torch function, this PR will improve timm models performance a lot.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/mkldnn_utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import torch
+
+# check the pattern: (nn.module, F.function) matched.
+def matches_module_function_pattern(pattern, node, modules):
```
Comment (jgong5) [inline torch/_inductor/mkldnn_utils.py:4]: It is a general function, not mkldnn specific? Why do we move it to mkldnn_util.py?

Comment (XiaobingSuper) [reply torch/_inductor/mkldnn_utils.py:4]: Yes, how about fx_utils.py?

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/mkldnn.py [hunk 2]
```diff
@@ -510,21 +513,50 @@ def mkldnn_fuse_fx(gm: torch.fx.GraphModule, example_inputs):
...
                 computation_node = modules[node.args[0].target]
-                unary_node = modules[node.target]
+                if node.op == "call_function":
+                    # make sure unary function's inputs only one fx.node(others should be constant value).
+                    if any(isinstance(v, torch.fx.Node) for v in node.args[1:]) or any(
+                        isinstance(v, torch.fx.Node) for _, v in node.kwargs.items()
+                    ):
+                        continue
+                    unary_node = create_unary_module(node)
+                    unary_node.eval()
```
Comment (desertfire) [inline torch/_inductor/mkldnn.py:557]: Not sure if it will ever happen, but do we need to check if `node` is in `training` before calling `eval()` here?

Comment (jgong5) [reply torch/_inductor/mkldnn.py:557]: Good catch. I guess it is needed. A similar check was applied after the code at here too: 
https://github.com/pytorch/pytorch/pull/91609/files#diff-7c920a79e1ea5c56baef3ef14306bc2a3a4eb4a13ca7689d93e0549eccd816d4R560

Comment (XiaobingSuper) [reply torch/_inductor/mkldnn.py:557]: For torch function, there doesn't have **training ** attr, even for **nn.module** case of that activation, the **training ** attr is not working, so I think we don't need to do such a check here.

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**